wolfcrypt/src/wc_port.c

```diff
@@ -1076,6 +1081,48 @@ size_t wc_strlcat(char *dst, const char *src, size_t dstSize)
 }
 #endif /* USE_WOLF_STRLCAT */
 
+#ifdef USE_WOLF_STRCASECMP
+int wc_strcasecmp(const char *s1, const char *s2)
+{
+    char c1, c2;
+    for (;
+         ;
+         ++s1, ++s2)
+    {
+        c1 = *s1;
+        if ((c1 >= 'a') && (c1 <= 'z'))
+            c1 -= ('a' - 'A');
+        c2 = *s2;
+        if ((c2 >= 'a') && (c2 <= 'z'))
+            c2 -= ('a' - 'A');
+        if ((c1 != c2) || (c1 == 0))
```

Comment:
c2 == 0 check too?

Reply:
we don't need `c2 == 0` because we already know `c1 != c2` and `c1 != 0`, i.e. `c2` can't be `0` either.